test/conformance/helpers/broker_control_plane_test_helper.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/eventing/pkg/apis/eventing"
+	eventingv1beta1 "knative.dev/eventing/pkg/apis/eventing/v1beta1"
+
+	"knative.dev/eventing/test/lib"
+	"knative.dev/eventing/test/lib/duck"
+	"knative.dev/eventing/test/lib/resources"
+)
+
+func BrokerV1Beta1ControlPlaneTestHelperWithChannelTestRunner(
+	t *testing.T,
+	brokerClass string,
+	channelTestRunner lib.ChannelTestRunner,
+	setupClient ...lib.SetupClientOption,
+) {
+	channelTestRunner.RunTests(t, lib.FeatureBasic, func(t *testing.T, channel metav1.TypeMeta) {
+		client := lib.Setup(t, true, setupClient...)
+		defer lib.TearDown(client)
+
+		t.Run("Trigger V1Beta1 can be crated before Broker (with attributes filter)", func(t *testing.T) {
+			TriggerV1Beta1BeforeBrokerHelper(t, brokerClass, client, channel)
+		})
+
+		t.Run("Broker V1Beta1 can be created and progresses to Ready", func(t *testing.T) {
+			BrokerV1Beta1CreatedToReadyHelper(t, brokerClass, client, channel)
+		})
+
+		t.Run("Ready Broker V1Beta1 is Addressable", func(t *testing.T) {
+			ReadyBrokerV1Beta1AvailableHelper(t, brokerClass, client, channel)
+		})
+
+		t.Run("Trigger V1Beta1 with Ready broker progresses to Ready", func(t *testing.T) {
+			TriggerV1Beta1ReadyBrokerReadyHelper(t, brokerClass, client, channel)
+		})
+
+		t.Run("Ready Trigger V1Beta1 includes status.subscriber Uri", func(t *testing.T) {
+			TriggerV1Beta1ReadyIncludesSubURI(t, brokerClass, client, channel)
+		})
+	})
+
+}
+
+func TriggerV1Beta1BeforeBrokerHelper(t *testing.T, brokerClass string, client *lib.Client, channel metav1.TypeMeta) {
+	const etLogger = "logger"
+	const loggerPodName = "logger-pod"
+
+	logPod := resources.EventRecordPod(loggerPodName)
+	client.CreatePodOrFail(logPod, lib.WithService(loggerPodName))
+	client.WaitForAllTestResourcesReadyOrFail() //Can't do this for the trigger because it's not 'ready' yet
+	client.CreateTriggerOrFail("trigger-no-broker",
+		resources.WithAttributesTriggerFilter(eventingv1beta1.TriggerAnyFilter, etLogger, map[string]interface{}{}),
+		resources.WithSubscriberServiceRefForTrigger(loggerPodName),
+	)
+	client.Tracker.Clean(true)
```

Comment:
Do we need this here? Is there any reason that we want to clean up the test resources before they're cleaned up with `defer lib.TearDown(client)`?

Reply:
Checking if the `trigger-no-broker` trigger progresses to Ready after the broker has been created and then check its `SubscriberURI` would be an interesting scenario to test.

Reply:
@aliok  good catch -- dropped (originally had it because I was trying to `client.WaitForAllTestResourcesReadyOrFail()` but eventually went with more specific wait's

@pierDipi sure -- despite not being (explicitly) in the spec I think that'd be a great add to ensure the case is covered, added.